egui/src/ui.rs

```diff
@@ -1108,6 +1108,22 @@ impl Ui {
         Button::new(text).ui(self)
     }
 
+    /// Usage: `if ui.button_with_image(texture_id, size, "Click me").clicked() { … }`
+    ///
+    /// Shortcut for `add(Button::image_and_text(texture_id, size, text))`
+    ///
+    /// See also [`Button`].
+    #[must_use = "You should check if the user clicked this with `if ui.button_with_image(…).clicked() { … } "]
+    #[inline(always)]
+    pub fn button_with_image(
+        &mut self,
+        texture_id: TextureId,
+        size: impl Into<Vec2>,
+        text: impl ToString,
+    ) -> Response {
+        Button::image_and_text(texture_id, size, text).ui(self)
+    }
```

Comment:
I'm not sure `Button::image_and_text` will be used enough to warrant a helper in `Ui`. I try to limit that to only the most commonly used things (there isn't even a `ui.slider`!)

Reply:
Ok, I've removed this one